common/libraries/crypto/mpz_operations/mpz_ecdsa.c

```diff
@@ -0,0 +1,288 @@
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "mpz_ecdsa.h"
+
+#include "mini-gmp.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+void mpz_curve_point_init(mpz_curve_point *p) {
+  mpz_init(p->x);
+  mpz_init(p->y);
+}
+
+void mpz_curve_point_clear(mpz_curve_point *p) {
+  mpz_clear(p->x);
+  mpz_clear(p->y);
+}
+
+// Set cp2 = cp1
+void mpz_curve_point_copy(const mpz_curve_point *cp1, mpz_curve_point *cp2) {
+  mpz_curve_point_init(cp2);
+
+  mpz_set(cp2->x, cp1->x);
+  mpz_set(cp2->y, cp1->y);
+}
+
+void mpz_curve_point_add(const mpz_curve *curve,
+                         const mpz_curve_point *cp1,
+                         mpz_curve_point *cp2) {
+  mpz_t lambda, inv, xr, yr;
+
+  mpz_init(lambda);
+  mpz_init(inv);
+  mpz_init(xr);
+  mpz_init(yr);
+
+  if (mpz_curve_point_is_infinity(cp1)) {
+    return;
+  }
+  if (mpz_curve_point_is_infinity(cp2)) {
+    mpz_curve_point_copy(cp1, cp2);
```

Comment:
This function re-initializes `cp2`, that will cause dangling pointers if `cp2` is already initialized

Reply:
noted, will fix it; 
also re-initialization would be causing memory leaks rather than dangling pointers?